apps/openmw/mwmechanics/spellcasting.cpp

```diff
@@ -1149,6 +1123,27 @@ namespace MWMechanics
             break;
         }
 
+        case ESM::MagicEffect::CurePoison:
+            actor.getClass().getCreatureStats(actor).getActiveSpells().purgeEffect(ESM::MagicEffect::Poison);
+            break;
+        case ESM::MagicEffect::CureParalyzation:
+            actor.getClass().getCreatureStats(actor).getActiveSpells().purgeEffect(ESM::MagicEffect::Paralyze);
+            break;
+        case ESM::MagicEffect::CureCommonDisease:
+            actor.getClass().getCreatureStats(actor).getSpells().purgeCommonDisease();
+            break;
+        case ESM::MagicEffect::CureBlightDisease:
+            actor.getClass().getCreatureStats(actor).getSpells().purgeBlightDisease();
+            break;
+        case ESM::MagicEffect::CureCorprusDisease:
+            actor.getClass().getCreatureStats(actor).getSpells().purgeCorprusDisease();
+            break;
+        case ESM::MagicEffect::Dispel:
+            actor.getClass().getCreatureStats(actor).getActiveSpells().purgeAll(magnitude);
```

Comment:
I just tested in vanilla, and it does apply over a duration there. You can see it with togglemagicstats (tms) and verify by casting it and then having a dispel-affected effect be applied to you within the duration to you. For example, have a non-actor cast "ash feast" (drain health) on you. It will be dispelled even though it is after the cast. So we should implement it that way I would think. If you want to fix it feel free, or I'll give it a shot when I have some time.